### ISSUE TITLE
WRONG_MULTIPLE_MODIFIERS_ORDER triggers on SAM interface

### ISSUE THREAD

**Issue by illarionov** [saveourtool/diktat#1598]
## Describe the bug

Having the following valid code snippet:
```kotlin
public fun interface Factory {
    public fun create(): List<Int>
}
```

Diktat fails with the error:
```
[WRONG_MULTIPLE_MODIFIERS_ORDER] sequence of modifier-keywords is incorrect: public should be on position 2, but is on position 1 (diktat-ruleset:multiple-modifiers)
[WRONG_MULTIPLE_MODIFIERS_ORDER] sequence of modifier-keywords is incorrect: fun should be on position 1, but is on position 2 (diktat-ruleset:multiple-modifiers
```

## Expected behavior
`public fun interface` should not trigger this rule

## Environment information
* diktat version: diktat-1.2.4.2.jar
* how is diktat run (CLI, plugin, etc.): CLI
* kotlin version: 1.8.0

**Comment by akuleshov7** [comment]
Thank you for your report! Yes, we missed it actually. Also not mentioned in our code style

**Comment by akuleshov7** [comment]
@illarionov how can I have missed that? 😄 When SAM interfaces were supported?
